base/compiler/methodtable.jl

```diff
@@ -40,45 +40,36 @@ end
 getindex(result::MethodLookupResult, idx::Int) = getindex(result.matches, idx)::MethodMatch
 
 """
-    findall(sig::Type, view::MethodTableView; limit=typemax(Int))
+    findall(sig::Type, view::MethodTableView; limit::Int=typemax(Int)) -> MethodLookupResult or missing
 
 Find all methods in the given method table `view` that are applicable to the
 given signature `sig`. If no applicable methods are found, an empty result is
 returned. If the number of applicable methods exceeded the specified limit,
 `missing` is returned.
 """
 function findall(@nospecialize(sig::Type), table::InternalMethodTable; limit::Int=typemax(Int))
-    _min_val = RefValue{UInt}(typemin(UInt))
-    _max_val = RefValue{UInt}(typemax(UInt))
-    _ambig = RefValue{Int32}(0)
-    ms = _methods_by_ftype(sig, nothing, limit, table.world, false, _min_val, _max_val, _ambig)
-    if ms === false
-        return missing
-    end
-    return MethodLookupResult(ms::Vector{Any}, WorldRange(_min_val[], _max_val[]), _ambig[] != 0)
+    return _findall(sig, nothing, table.world, limit)
 end
 
 function findall(@nospecialize(sig::Type), table::OverlayMethodTable; limit::Int=typemax(Int))
+    result = _findall(sig, table.mt, table.world, limit)
+    result === missing || isempty(result) || return result
```

Comment:
If this part of the result is `missing`, then the total result must also be `missing`

Though I find the change between `false` and `missing` to be somewhat odd for the API. Perhaps we should use something consistent such as `nothing`? (even `true` might be a better choice, since it means we found too many matches)

Reply:
Yeah, I think `nothing` or `missing` would be better, since they are easier to be handled by `Conditional`? Given that such case means "too many matches", `missing` sounds a bit counter-intuitive.

Reply:
https://github.com/JuliaLang/julia/pull/44478